aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -90,13 +83,20 @@ impl<'a, S: 'a + StateView + Sync> ExecutorTask for AptosExecutorTask<'a, S> {
                     ExecutionStatus::DelayedFieldsCodeInvariantError(
                         vm_status.message().cloned().unwrap_or_default(),
                     )
-                } else if AptosVM::should_restart_execution(&vm_output) {
+                } else if AptosVM::should_restart_execution(vm_output.change_set()) {
                     speculative_info!(
                         &log_context,
                         "Reconfiguration occurred: restart required".into()
                     );
                     ExecutionStatus::SkipRest(AptosTransactionOutput::new(vm_output))
                 } else {
+                    if (executor_with_group_view.is_delayed_field_optimization_capable()
+                        || executor_with_group_view.is_resource_group_split_in_change_set_capable())
+                        && !Self::is_transaction_dynamic_change_set_capable(txn)
+                    {
+                        return ExecutionStatus::DirectWriteSetTransactionNotCapableError;
```

Comment:
If we never return it, can we get rid of `DirectWriteSetTransactionNotCapableError` completely?

Reply:
we would return it if VM doesn't create SkipRest on these transactions.

do you mean removing from error status to assert instead?

Reply:
yes